### PR TITLE
enabling extent reports support

### PR DESCRIPTION
**Pull request by ManjulaPaluru**
enabling extent reports support

### CODE REVIEW

File: src/main/java/com/qa/opencart/factory/DriverFactory.java [hunk 1]
```diff
@@ -1,52 +1,79 @@
...
 import java.io.IOException;
 import java.util.Properties;
 
 public class DriverFactory {
+
     public WebDriver driver;
     public Properties prop;
+    public static String highlight;
+    public OptionsManager optionsManager;
+    public static ThreadLocal<WebDriver> tlDriver = new ThreadLocal<WebDriver>();
```
Comment (ssaguturu) [inline src/main/java/com/qa/opencart/factory/DriverFactory.java:24]: Why are you using `public` variables?

File: src/main/java/com/qa/opencart/factory/DriverFactory.java [hunk 2]
```diff
@@ -1,52 +1,79 @@
...
         String browserName = prop.getProperty("browser").trim();
-        System.out.println("browserName is: " + browserName);
+        System.out.println("browser name is : " + browserName);
+        highlight = prop.getProperty("highlight").trim();
+        optionsManager = new OptionsManager(prop);
+
         if (browserName.equalsIgnoreCase("chrome")) {
             WebDriverManager.chromedriver().setup();
-            driver = new ChromeDriver();
+            // driver = new ChromeDriver(optionsManager.getChromeOptions());
```
Comment (ssaguturu) [inline src/main/java/com/qa/opencart/factory/DriverFactory.java]: Please remove commented code

File: src/main/java/com/qa/opencart/factory/DriverFactory.java [hunk 3]
```diff
@@ -60,4 +87,21 @@ public Properties init_prop() {
...
+     * take screenshot
+     */
+    public static String getScreenshot() {
+        File srcFile = ((TakesScreenshot)getDriver()).getScreenshotAs(OutputType.FILE);
+        String path = System.getProperty("user.dir")+"/screenshot/"+System.currentTimeMillis()+".png";
+        File destination = new File(path);
+        try {
+            FileUtils.copyFile(srcFile, destination);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ssaguturu) [inline src/main/java/com/qa/opencart/factory/DriverFactory.java:97]: Can you use `log4j` to log exceptions, and messages?

File: src/main/java/com/qa/opencart/listeners/ExtentReportListener.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		 */
+		extentTest.assignCategory(className);
+		test.set(extentTest);
+		test.get().getModel().setStartTime(getTime(result.getStartMillis()));
+	}
+
+	public synchronized void onTestSuccess(ITestResult result) {
+		System.out.println((result.getMethod().getMethodName() + " passed!"));
+		test.get().pass("Test passed");
+		//test.get().pass(result.getThrowable(), MediaEntityBuilder.createScreenCaptureFromPath(getScreenshot()).build());
```
Comment (ssaguturu) [inline src/main/java/com/qa/opencart/listeners/ExtentReportListener.java:91]: Don't you need screenshot for `onSuccess`?

File: src/main/java/com/qa/opencart/utils/Constants.java [hunk 5]
```diff
@@ -1,7 +1,6 @@
 package com.qa.opencart.utils;
 
 import edu.emory.mathcs.backport.java.util.Arrays;
-import org.apache.commons.collections4.functors.NullIsTruePredicate;
```
Comment (ssaguturu) [inline src/main/java/com/qa/opencart/utils/Constants.java:4]: Thanks for removing unused imports

File: src/main/java/com/qa/opencart/utils/JavaScriptUtil.java [hunk 6]
```diff
@@ -3,84 +3,84 @@
...
-            } catch (InterruptedException e) {
-            }
-        }
-
-        public String getTitleByJS() {
-            JavascriptExecutor js = (JavascriptExecutor) driver;
-            return js.executeScript("return document.title;").toString();
+        try {
+            Thread.sleep(20);
+        } catch (InterruptedException e) {
```
Comment (ssaguturu) [inline src/main/java/com/qa/opencart/utils/JavaScriptUtil.java:27]: `catch` block without anything is not a good practice

File: src/test/java/com/qa/opencart/base/BaseTest.java [hunk 7]
```diff
@@ -26,7 +26,7 @@ public void setup(){
         System.out.println("@BeforeTest");
         df=new DriverFactory();
         prop=df.init_prop();
-       driver= df.init_driver(/*prop*/);
+       driver= df.init_driver(prop);
```
Comment (ssaguturu) [inline src/test/java/com/qa/opencart/base/BaseTest.java:29]: I still don't understand, why you need the `prop` parameter.

**Review by ssaguturu — APPROVED** [review]

File: src/main/java/com/qa/opencart/factory/DriverFactory.java (hunk 1 above)
Comment (ManjulaPaluru) [reply src/main/java/com/qa/opencart/factory/DriverFactory.java:24]: For accessing variables publicly

Comment (ssaguturu) [reply src/main/java/com/qa/opencart/factory/DriverFactory.java:24]: ha ha!!! 🙉

**Review by ssaguturu — APPROVED** [review]

**Pull request merged**